task11/Shipment.ts

```diff
@@ -0,0 +1,105 @@
+import { AirEeastShipper, ChicagoSprintShipper, PacificParcelShipper, Shipper } from "./Shipper";
+import { State } from "./State";
+
+let uniqueId = 1;
+
+export enum ShipmentType {
+    Package,
+    Letter,
+    Oversize,
+}
+
+export interface IShipment {
+    // how to hide these props? currently only need in the Shipment class
+    state: State;
+    shipper: Shipper;
+    shipmentType: ShipmentType;
+
+    addMark(mark?: string): void;
+    ship(): string;
+    getShipmentId(): string
```

Comment:
You don't really need so many things here
You could just leave functions which you will use in Decorator as well
```
    addMark(mark?: string): void;
    ship(): string;
```
 As discussed on the call `addMark(mark?: string): void; `should be replaced with a function wich helps to get arrays with all the marks